packages/flutter/lib/src/cupertino/refresh.dart

```diff
@@ -0,0 +1,514 @@
+// Copyright 2018 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+import 'dart:async';
+import 'dart:math';
+
+import 'package:flutter/foundation.dart';
+import 'package:flutter/rendering.dart';
+import 'package:flutter/scheduler.dart';
+import 'package:flutter/services.dart';
+import 'package:flutter/widgets.dart';
+import 'activity_indicator.dart';
+import 'colors.dart';
+import 'icons.dart';
+
+class _CupertinoRefreshSliver extends SingleChildRenderObjectWidget {
+  const _CupertinoRefreshSliver({
+    this.refreshIndicatorLayoutExtent: 0.0,
+    this.hasLayoutExtent,
+    Widget child,
+  }) : super(child: child);
+
+  /// The amount of space the indicator should occupy in the sliver in a
+  /// resting state when in the refreshing mode.
+  final double refreshIndicatorLayoutExtent;
+  /// _RenderCupertinoRefreshSliver will paint the child in the available
+  /// space either way but this instructs the _RenderCupertinoRefreshSliver
+  /// on whether to also occupy any layoutExtent space or not.
+  final bool hasLayoutExtent;
+
+  @override
+  _RenderCupertinoRefreshSliver createRenderObject(BuildContext context) {
+    return new _RenderCupertinoRefreshSliver(
+      refreshIndicatorExtent: refreshIndicatorLayoutExtent,
+      hasLayoutExtent: hasLayoutExtent,
+    );
+  }
+
+  @override
+  void updateRenderObject(BuildContext context, covariant _RenderCupertinoRefreshSliver renderObject) {
+    renderObject
+        ..refreshIndicatorLayoutExtent = refreshIndicatorLayoutExtent
+        ..hasLayoutExtent = hasLayoutExtent;
```

Comment:
indent by two fewer spaces for consistency

Reply:
Ummm I just realized https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo doesn't say to use 4 spaces indents anywhere. Wonder if it changed recently. Done.